src/Service/ClientServiceFactory.php

```diff
@@ -14,7 +14,11 @@ class ClientServiceFactory
     public function __invoke($container)
     {
         $config     = $container->get('config');
-        $httpClient = new HttpClient();
+        $httpClient = new HttpClient(
+            null,
+            !empty($config['apigility-consumer']['api-http-adapter-options']) ?
```

Comment:
as curl only supposed to be used for digest, it should only happen in `ClientService::callAPI()` using `$this->httpClient->setOptions()`.